doc/develop/coding-guidelines.rst

```diff
@@ -212,3 +212,53 @@ For e.g. 4.1 standard and internal imports
                                 MD_AT_TEMPSPECTRUM, MD_AT_FIB)
 
 
+Naming Convention for Pull Requests, Branches and Commits
+==========================================================
+
+* The pull request title and the branch name should be the same.
+
+    * The words in a pull request title should be separated by spaces, for the branch name they should be separated by dashes.
+    * Recommended naming convention: *[ticket] title*.
+    * It is recommended to use the *[]* only for the pull request name.
+|
+* A pull request should contain a minimal description of the changes and what problem they solve.
+
+    * One can include images, links, and tables to help convey this information.
+    * If a pull request contains a single commit it is recommended to use the commit message as the pull request description.
+    * Example template of a pull request description:
```

Comment:
Is the example template always need to be followed or is it a "good to follow"? Little clarification on it will be nice